components/collector/src/source_collectors/dependency_track/source_up_to_dateness.py

```diff
@@ -36,17 +36,40 @@ async def _parse_entities(self, responses: SourceResponses) -> Entities:
         entities = Entities()
         for response in responses:
             async for project in self._get_projects_from_response(response):
+                last_bom_analysis = self._last_bom_analysis_datetime(project)
+                last_bom_import = self._last_bom_import_datetime(project)
                 uuid = project["uuid"]
                 entity = Entity(
                     key=uuid,
-                    last_bom_import=self._last_bom_import_datetime(project).isoformat(),
+                    last_bom_analysis="" if last_bom_analysis is None else last_bom_analysis.isoformat(),
+                    last_bom_import="" if last_bom_import is None else last_bom_import.isoformat(),
                     project=project["name"],
                     project_landing_url=f"{landing_url}/projects/{uuid}",
                 )
                 entities.append(entity)
         return entities
 
-    @staticmethod
-    def _last_bom_import_datetime(project: DependencyTrackProject) -> datetime:
+    def _project_event_datetimes(self, project: DependencyTrackProject) -> list[datetime]:
+        """Return the project event (BOM import, BOM analysis) datetimes."""
+        datetimes = []
+        event_types = self._parameter("project_event_types")
+        if "last BOM import" in event_types and (datetime := self._last_bom_import_datetime(project)):
+            datetimes.append(datetime)
+        if "last BOM analysis" in event_types and (datetime := self._last_bom_analysis_datetime(project)):
+            datetimes.append(datetime)
+        return datetimes
+
+    @classmethod
+    def _last_bom_import_datetime(cls, project: DependencyTrackProject) -> datetime | None:
         """Return the last BOM import datetime for the project."""
-        return datetime_from_timestamp(int(project["lastBomImport"]))
+        return cls._timestamp_to_datetime(project.get("lastBomImport"))
+
+    @classmethod
+    def _last_bom_analysis_datetime(cls, project: DependencyTrackProject) -> datetime | None:
+        """Return the last BOM analysis datetime for the project."""
+        return cls._timestamp_to_datetime(project["metrics"].get("lastOccurrence"))
```

Comment:
```suggestion
        return cls._timestamp_to_datetime(project.get("metrics", {}).get("lastOccurrence"))
```

Reply:
A newly created (empty) project does not have `metrics` in the project dict, if needed for test case:
```json
{
	"1": {
		"name": "test",
		"classifier": "APPLICATION",
		"uuid": "d31c5b4f-881e-4e95-8d56-f188fe5f6d95",
		"children": [],
		"properties": [],
		"tags": [],
		"active": true
	}
}
```